dali/test/python/test_operator_readers_webdataset_requirements.py

```diff
@@ -0,0 +1,427 @@
+    index_file = generate_temp_index_file(tar_file_path)
+
+    extract_dir = generate_temp_extract(tar_file_path)
+    equivalent_files = sorted(
+        glob(extract_dir.name + "/*"), key=lambda s: int(s[s.rfind("/") + 1 : s.rfind(".")])
+    )
+
+    compare_pipelines(
+        webdataset_raw_pipeline(
+            tar_file_path,
+            index_file.name,
+            ["jpg", "txt"],
+            missing_component_behavior="fillempty",
+            batch_size=test_batch_size,
+            device_id=0,
+            num_threads=1,
+        ),
+        file_reader_pipeline(
+            equivalent_files, ["jpg", []], batch_size=test_batch_size, device_id=0, num_threads=1
+        ),
+        test_batch_size,
+        math.ceil(num_samples / test_batch_size),
+    )
+
+
+def test_skip_sample():
+    global test_batch_size
+    num_samples = 500
+    tar_file_path = get_dali_extra_path() + "/db/webdataset/MNIST/missing.tar"
+    index_file = generate_temp_index_file(tar_file_path)
+
+    extract_dir = generate_temp_extract(tar_file_path)
+    equivalent_files = list(
+        filter(
+            lambda s: int(s[s.rfind("/") + 1 : s.rfind(".")]) < 2500,
+            sorted(
+                glob(extract_dir.name + "/*"), key=lambda s: int(s[s.rfind("/") + 1 : s.rfind(".")])
+            ),
+        )
+    )
+
+    compare_pipelines(
+        webdataset_raw_pipeline(
+            tar_file_path,
+            index_file.name,
+            ["jpg", "cls"],
+            missing_component_behavior="skip",
+            batch_size=test_batch_size,
+            device_id=0,
+            num_threads=1,
+        ),
+        file_reader_pipeline(
+            equivalent_files, ["jpg", "cls"], batch_size=test_batch_size, device_id=0, num_threads=1
+        ),
+        test_batch_size,
+        math.ceil(num_samples / test_batch_size),
+    )
+    wds_pipeline = webdataset_raw_pipeline(
+        tar_file_path,
+        index_file.name,
+        ["jpg", "cls"],
+        missing_component_behavior="skip",
+        batch_size=test_batch_size,
+        device_id=0,
+        num_threads=1,
+    )
+    wds_pipeline.build()
+    assert_equal(list(wds_pipeline.epoch_size().values())[0], num_samples)
+
+
+def test_raise_error_on_missing():
+    global test_batch_size
+    num_samples = 1000
+    tar_file_path = get_dali_extra_path() + "/db/webdataset/MNIST/missing.tar"
+    index_file = generate_temp_index_file(tar_file_path)
+
+    extract_dir = generate_temp_extract(tar_file_path)
+    equivalent_files = sorted(
+        glob(extract_dir.name + "/*"), key=lambda s: int(s[s.rfind("/") + 1 : s.rfind(".")])
+    )[:num_samples]
+
+    wds_pipeline = webdataset_raw_pipeline(
+        tar_file_path,
+        index_file.name,
+        ["jpg", "cls"],
+        missing_component_behavior="error",
+        batch_size=test_batch_size,
+        device_id=0,
+        num_threads=1,
+    )
+    assert_raises(RuntimeError, wds_pipeline.build)
```

Comment:
Please use nose_utils and check the error message as well.

Reply:
alright